src/renderer/components/ft-element-list/ft-element-list.js

```diff
@@ -18,12 +19,85 @@ export default Vue.extend({
   },
   data: function () {
     return {
-      test: 'hello'
+      test: 'hello',
+      re: /^#\/($|subscriptions|channel|userplaylists|history)/
     }
   },
   computed: {
     listType: function () {
       return this.$store.getters.getListType
+    },
+
+    channelBlockerList: function () {
+      return this.$store.getters.getChannelBlockerList
+    },
+
+    avoidChannelBlockerByURI: function () {
+      // Won't hide:
+      //  Subscriptions (#/ or #/subscriptions)
+      //  Channel page (#/channel/UCuAXFkgsw1L7xaCfnd5JJOw)
+      //  Your playlists (#/userplaylists)
+      //  History (#/history)
+      const match = window.location.hash.match(this.re)
+      if (match === null) {
+        return false
+      }
+      return true
     }
+  },
+  methods: {
+    isChannelBlocked: function (video) {
+      const channelIndex = this.channelBlockerList.findIndex((item) => {
+        return item.authorId === video.authorId
+      })
+
+      return channelIndex !== -1
+    },
+
+    toggleBlockedChannel: function (channel) {
+      if (this.isChannelBlocked(channel)) {
+        this.removeChannelFromBlockList(channel)
+      } else {
+        this.addChannelToBlockList(channel)
+      }
+    },
+
+    addChannelToBlockList: function (channel) {
+      // channel: {authorId: 'UCuAXFkgsw1L7xaCfnd5JJOw', author: 'Rick Astley'}
+      console.log('adding channel', JSON.stringify(channel))
+
+      const newList = this.channelBlockerList.slice()
+      newList.push(channel)
+      newList.sort((a, b) => {
+        return a.author.localeCompare(b.author)
```

Comment:
I think you should pass in the user's locale into localeCompare. Ex: https://github.com/FreeTubeApp/FreeTube/blob/3b195c3ad58aa51528edeadde210ea5c54cc7ad2/src/renderer/components/watch-video-info/watch-video-info.js#L207